### PR TITLE
[Drata] Part 2 - New Actions

### PR DESCRIPTION
**Pull request by andrewjschuang**
Part 2 out 3 of #5139.

Actions:

- [X] Create Asset
- [X] Create Control
- [X] Create Vendor
- [X] Find Controls
- [X] Find Personnel
- [X] Find Vendors
- [X] Update Vendor
- [X] Upload Background Check
- [X] Upload Device Compliance Document
- [X] Upload Evidence
- [X] Upload User Compliance Document

**Note for PR Reviewer:** In this PR, please review only the files in `actions/` and `drata.app.mjs`.

### <samp>🤖 Generated by Copilot at 12c1dd7</samp>

This pull request adds new action and source components for the Drata app, which is a compliance automation platform. The action components allow users to create, update, find, and upload different resources in Drata, such as assets, controls, vendors, personnel, and documents. The source components emit events for new resources added in Drata, such as assets, controls, evidences, and personnel. The pull request also adds common files for the Drata app definition, constants, and utilities, and updates the `package.json` file. The pull request deletes the unused `.gitignore` file.

### <samp>🤖 Generated by Copilot at 12c1dd7</samp>

> _`Drata` components_
> _Actions, sources, and helpers_
> _Autumn of compliance_

### <samp>🤖 Generated by Copilot at 12c1dd7</samp>

*  Add new action components to create, update, find, and upload resources in Drata ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-c86e1662c8e6d8ed39e41867f186a58ee916f671d695a798086c46bb3c1db5bcR1-R105), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-5d59c9a53ce9ac90d972c2a7ffeeea7ef1e5c0f52395ccac5445e6d828a026f4R1-R62), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-6bad81e2ede0f4e8c3e655e9a886b941a02d5b50b2b087a6eabc4595a1170eb0R1-R219), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-ef022ab6f2cd79eb416d8709ab76cfad33d9bc73b76e5a40367d3e63d84440c0R1-R138), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-6fc3124e161dfdbfcbd3ba77757250d1a89eb68343fd24d7d68f931a1a9776e3R1-R142), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-cc630ee99b563f077851b2d43e2e653600356339ed85694c9fa96db2429fcbe0R1-R225), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-650a63536f174fe90eba52f7955560ec822dac79d90a63a4bf0cd3b984e63c20R1-R56), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-60b002040098c47c885a597dacd638b9ee7ca5bc763e353ae519b547e798da02R1-R111), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-eddcddc1253db63df45f9fa1042bd6e7f6e3041cf783a713fd1839f5430853ddR1-R111))
* Add new source components to emit events for new resources added in Drata ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-cae736d151093a9eb01cf81fb7fb4248c39c6dbc2dd9c274397792585de1e21bR1-R86), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-6449b3851ac7649bb15eb4ece7ee9c9702cf24eb5f13950a31546dd2f2815eabR1-R75), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-72dde9f7b6c9163c9da845c99c149ef3a65a29d495728debe281645284e35370R1-R84), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-ccca2dc6f2723dae67ff252148831b75165fd625be808082e3e7248e2d47decfR1-R111))
* Add a new file with the Drata app definition, which includes the type, app, prop definitions, and methods ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-a07058fe354e5714b5b288324099660f299a6c49cb5ec445a7251a3072c43db2R1-R269))
* Add a new file with common constants used by the Drata components, such as the async options limit, the deploy limit, the pagination limit, and the sort criteria ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-ad40b68422e9aa2a165ae2c416fc624458a76e16b6e166ea2bd7cf215e4d56f0R1-R10))
* Add a new file with common utility methods used by the Drata components, such as getting the personnel name, initializing the JSON props, and downloading a file ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-557f97192f075cec88be2ca98c4d90a2f89af57788470577936f1a6f2b9b4106R1-R22))
* Update the `package.json` file for the Drata components, which includes the name, version, description, main, keywords, homepage, author, publish config, and dependencies ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-f308bcfbd8557663c1c9360bb4d8bc09ad1946ad1dfe5c5bc638abe738126f45L3-R17))
* Delete the unused files `components/drata/.gitignore` and `components/drata/app/drata.app.ts` ([link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-7506fdc3e6d938478e697eeeb6ce05845536e1a978ede9657834c9549fdb79d0), [link](https://github.com/PipedreamHQ/pipedream/pull/6027/files?diff=unified&w=0#diff-27405f52b709f4496e9f16016505e167d7bc6d11623b4451d89abcea14a553d5))

### CODE REVIEW

File: components/drata/actions/create-asset/create-asset.mjs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      type: "string",
+      label: "Asset Type",
+      description: "The asset type",
+      options: [
+        "PHYSICAL",
+        "VIRTUAL",
+      ],
+    },
+    ownerId: {
+      type: "integer",
```
Comment (dylburger) [inline components/drata/actions/create-asset/create-asset.mjs:51]: Does Drata provide an API for listing owners?

File: components/drata/actions/create-asset/create-asset.mjs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      description: "Notes about the asset",
+      optional: true,
+    },
+    uniqueId: {
+      type: "string",
+      label: "Unique ID",
+      description: "Unique Id associated with this asset",
+      optional: true,
+    },
+    removedAt: {
```
Comment (dylburger) [inline components/drata/actions/create-asset/create-asset.mjs:67]: Can you note the required date format on all the date props? We really need an ISO string native type with a date picker for static values!

Screenshot 2023-04-18 at 10 46 09 AM

File: components/drata/actions/create-asset/create-asset.mjs (hunk 1 above)
Comment (andrewjschuang) [reply components/drata/actions/create-asset/create-asset.mjs:51]: I didn't find one. Also user, owner, personnel, and vendors `id`s are all a mysterious gray area to me

File: components/drata/actions/create-asset/create-asset.mjs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import drata from "../../drata.app.mjs";
```
Comment (dylburger) [inline components/drata/actions/create-asset/create-asset.mjs:1]: Validated and working

File: components/drata/actions/create-control/create-control.mjs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import drata from "../../drata.app.mjs";
```
Comment (dylburger) [inline components/drata/actions/create-control/create-control.mjs:1]: Validated

File: components/drata/actions/create-vendor/create-vendor.mjs [hunk 5]
```diff
@@ -0,0 +1,219 @@
+import drata from "../../drata.app.mjs";
```
Comment (dylburger) [inline components/drata/actions/create-vendor/create-vendor.mjs:1]: Validated

File: components/drata/actions/find-controls/find-controls.mjs [hunk 6]
```diff
@@ -0,0 +1,529 @@
+import drata from "../../drata.app.mjs";
```
Comment (dylburger) [inline components/drata/actions/find-controls/find-controls.mjs:1]: lgtm

File: components/drata/actions/find-personnel/find-personnel.mjs [hunk 7]
```diff
@@ -0,0 +1,138 @@
+import drata from "../../drata.app.mjs";
```
Comment (dylburger) [inline components/drata/actions/find-personnel/find-personnel.mjs:1]: lgtm

File: components/drata/actions/update-vendor/update-vendor.mjs [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+  name: "Update Vendor",
+  description: `Update a Vendor. [See docs here.](${docsLink})`,
+  version: "0.0.1",
+  type: "action",
+  props: {
+    drata,
+    vendorId: {
+      type: "integer",
+      label: "Vendor ID",
+      description: "The ID of the vendor",
```
Comment (dylburger) [inline components/drata/actions/update-vendor/update-vendor.mjs:16]: Can you implement async options for this based on the [list vendors](https://developers.drata.com/docs/openapi/reference/operation/VendorsPublicController_listVendors/) endpoint?

File: components/drata/actions/upload-background-check/upload-background-check.mjs [hunk 9]
```diff
@@ -0,0 +1,56 @@
+import drata from "../../drata.app.mjs";
```
Comment (dylburger) [inline components/drata/actions/upload-background-check/upload-background-check.mjs:1]: confirmed

File: components/drata/actions/upload-device-compliance-document/upload-device-compliance-document.mjs [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+export default {
+  key: "drata-upload-device-compliance-document",
+  name: "Upload Device Compliance Document",
+  description: `Upload a device compliance document. [See docs here.](${docsLink})`,
+  version: "0.0.1",
+  type: "action",
+  props: {
+    drata,
+    deviceId: {
```
Comment (dylburger) [inline components/drata/actions/upload-device-compliance-document/upload-device-compliance-document.mjs:14]: Can you implement async options from https://developers.drata.com/docs/openapi/reference/operation/DevicePublicController_getDeviceDocuments/ ?

**Review by dylburger — COMMENTED** [review]
Just a few comments on async options that I believe Drata should support — if they don't, don't worry about if, we can keep the IDs for now.

Feel free to get the code approved / merge once you're done. Everything else looks great.

File: components/drata/actions/upload-device-compliance-document/upload-device-compliance-document.mjs (hunk 10 above)
Comment (andrewjschuang) [reply components/drata/actions/upload-device-compliance-document/upload-device-compliance-document.mjs:14]: This endpoint is to list the documents for a specified `deviceId`

File: components/drata/actions/create-asset/create-asset.mjs (hunk 1 above)
Comment (andrewjschuang) [reply components/drata/actions/create-asset/create-asset.mjs:51]: I applied `personnelId` to this one

**Pull request merged**